search/search_repository.go

```diff
@@ -307,6 +310,30 @@ func parseMap(queryMap map[string]interface{}, q *Query) {
 	}
 }
 
+func parseOptions(queryMap map[string]interface{}, q *Query) {
+	for key, val := range queryMap {
+		switch val.(type) {
+		case []interface{}:
+			if key == OPTS {
+				options := QueryOptions{}
+				for _, v := range val.([]interface{}) {
+					if o, ok := v.(map[string]interface{}); ok {
+						for k, vl := range o {
+							switch k {
+							case "parent-exists":
```

Comment:
Can't you just access the map like so without a for-loop?

```go
if v1, ok := o["parent-exists"]; ok {
  options.ParentExists = true
} else if v1, ok := o["tree-view"]; ok {
  options.TreeView = true
}
```

And btw. please use constants for those "parent-exists" and "tree-view" strings.

Reply:
The options are not really related. So, we cannot use `else if`. A switch within a loop would be better here instead of multiple if statements.

Reply:
Removed string literals. 5773bcc4ac627837440fe366d7b18110ffd27ebf

Reply:
@baijum My whole point was that you don't need the loop `for k, vl := range o {` ;)

Reply:
And I agree that it should not be an `else if` but a simple `if`.

Reply:
@kwk If I am going to use switch I need that for-loop, right?

Reply:
Please give me some time. I am looking into this now.

Reply:
I think the `$OPTS` value need not be a array of maps. Can I change it to a map?

Reply:
Done. ca614827336298f2ac423eefb67ed75b0e9f8e0b